src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -107,7 +107,7 @@ function ($deferred, $namespace, &$expiredIds) use ($getId) {
      *
      * @return AdapterInterface
      */
-    public static function createSystemCache($namespace, $defaultLifetime, $version, $directory, LoggerInterface $logger = null)
+    public static function createSystemCache(string $namespace, int $defaultLifetime, string $version, string $directory, LoggerInterface $logger = null)
```

Comment:
all the param phpdoc should be removed

Reply:
btw, I also think the method should not be in AbstractAdapter. It get's inherited to all adapters. But it not meant to be overwritten or called with late static binding. It's a plain factory method, independent of all the adapters.